### DISCUSSION TITLE
与home assistant 的整合讨论

### DISCUSSION THREAD

**Discussion by lewangdev** [FoloToy/folotoy-server-self-hosting#78] [Ideas]
目前的思路是参考 Teslamate，通过mqtt对接

**Comment by marvyn** [comment]
是否有可能增加一些跟 HA 平台的数据交互，达到控制 HA 或者查询 HA 系统状态的效果。
比如，可以设定某个角色专用于和 HA 进行互动。
可能能够实现以下场景：
1. 向兔子发语音，如“我想去洗澡了”或“我打算看一部电影”
2. 服务端通过 HA 的 API 查询家中卫生间的设备状态，将语音命令、相关设备名和状态（如 {"entity_id": "light.bathroom", "status": "on"}）发送到 OpenAI
3. OpenAI 根据命令和设备状态，结合预设的 Prompt 返回 {"action": [{"service": "light.turn_on", "entity_id": "light.bathroom"}, {"service": "fan.set_percentage", "entity_id": "fan.bathroom", "data": {"percent": 50}}]}
4. 服务端发送这个结果到 HA，HA 执行开灯、开换气扇的操作，并语音提示操作结果。

以上是一点初步的想法。
